lib/rust/parser/src/macros/built_in.rs

```diff
@@ -230,7 +230,7 @@ fn export_body<'s>(
 /// If-then-else macro definition.
 pub fn if_then_else<'s>() -> Definition<'s> {
     crate::macro_definition! {
-    ("if", everything(), "then", everything(), "else", everything()) if_body}
+    ("if", everything(), "then", everything(), "else", or(block(), many(not_block()))) if_body}
```

Comment:
I am not completely sure what is supposed to be a difference between `everything()` and `or(x,not(x))` - in binary logic `x or not x` is always true - e.g. close to `everything()`. Here we have `or(block(), many(not_block())))` which is almost `or(block(), not_block()))` - e.g. `x or not x`.

Reply:
The `many` is important! The difference is that this will either match a block at the beginning, or a sequence of non-block items--but unlike `everything`, it *won't* match a sequence of non-block items followed by a block.